modules/dasBidAdapter.js

```diff
@@ -0,0 +1,240 @@
+  }
+
+  // Custom parameters
+  if (params.customParams && typeof params.customParams === 'object') {
+    Object.assign(customParams, params.customParams);
+  }
+
+  const pageContext = params.pageContext;
+  if (pageContext) {
+    // Document URL override
+    if (pageContext.du) {
+      customParams.du = pageContext.du;
+    }
+
+    // Referrer override
+    if (pageContext.dr) {
+      customParams.dr = pageContext.dr;
+    }
+
+    // Document version
+    if (pageContext.dv) {
+      customParams.DV = pageContext.dv;
+    }
+
+    // Keywords
+    const keywords = getAllOrtbKeywords(
+      bidderRequest?.ortb2,
+      pageContext.keyWords,
+    );
+    if (keywords.length > 0) {
+      customParams.kwrd = keywords.join('+');
+    }
+
+    // Local capping
+    if (pageContext.capping) {
+      customParams.local_capping = pageContext.capping;
+    }
+
+    // Key values
+    if (pageContext.keyValues && typeof pageContext.keyValues === 'object') {
+      Object.entries(pageContext.keyValues).forEach(([key, value]) => {
+        keyValues[`kv${key}`] = value;
+      });
+    }
+  }
+
+  const du = customParams.du || deepAccess(bidderRequest, 'refererInfo.page');
+  const dr = customParams.dr || deepAccess(bidderRequest, 'refererInfo.ref');
+
+  if (du) customParams.du = du;
+  if (dr) customParams.dr = dr;
+
+  const dsaRequired = deepAccess(bidderRequest, 'ortb2.regs.ext.dsa.required');
+  if (dsaRequired !== undefined) {
+    customParams.dsainfo = dsaRequired;
+  }
+
+  return {
+    customParams,
+    keyValues,
+  };
+}
+
+function buildUserIds(customParams) {
+  const userIds = {};
+  if (customParams.lu) {
+    userIds.lu = customParams.lu;
+  }
+  if (customParams.aid) {
+    userIds.aid = customParams.aid;
+  }
+  return userIds;
+}
+
+function getNpaFromPubConsent(pubConsent) {
+  const params = new URLSearchParams(pubConsent);
+  return params.get('npa') == '1';
+}
+
+function buildOpenRTBRequest(bidRequests, bidderRequest) {
+  const { customParams, keyValues } = parseParams(
+    bidRequests[0].params,
+    bidderRequest,
+  );
+  const imp = bidRequests.map((bid, index) => {
+    const sizes = getAdUnitSizes(bid);
+    const imp = {
+      id: bid.bidId,
+      tagid: bid.params.slot,
+      secure: 1,
+    };
+
+    if (bid.mediaTypes?.banner) {
+      imp.banner = {
+        format: sizes.map((size) => ({
+          w: size[0],
+          h: size[1],
+        })),
+        pos: bid.params.slotSequence || 0,
+      };
+    }
+
+    return imp;
+  });
+
+  const request = {
+    id: bidderRequest.bidderRequestId,
+    imp,
+    site: {
+      id: customParams.site,
+      page: customParams.du,
+      ref: customParams.dr,
+      ext: {
+        area: customParams.area,
+      },
+      ...bidderRequest.ortb2.site,
+    },
+    device: bidderRequest.device,
+    user: {
+      ext: {
+        ids: buildUserIds(customParams),
+      },
+    },
+    ext: {
+      network: customParams.network,
```

Comment:
Potrzebujemy tego, skoro już jest w ścieżce?

Reply:
jest w ścieżce, ale nie w backendzie. Request jest pod /{network}/bid i proxujemy to pod /bid

Reply:
Sprawdziłem, jak to jest podpięte w Akceleratorze i faktycznie. Ale w sumie czemu wycinamy tam network ze ścieżki?